src/components/autosuggest-text/autossugest-text.tsx

```diff
@@ -0,0 +1,94 @@
+
+export interface AutosuggestTextProps {
+  responseData: SuggestionsFromQueryStringQuery["suggest"]["result"];
+  currentQ: string | undefined;
+  highlightedIndex: number;
+  getItemProps: UseComboboxPropGetters<Suggestion>["getItemProps"];
+  stringSuggestionAuthorText?: string;
+  stringSuggestionWorkText?: string;
+  stringSuggestionTopicText?: string;
+}
+
+interface SuggestionCreator {
+  __typename: "Creator";
+  name: string;
+}
+interface SuggestonSubject {
+  __typename: "Subject";
+  value: string;
+}
+interface SuggestionWork {
+  __typename: "Work";
+  title?: string | null;
+}
+export type Suggestion = SuggestionCreator | SuggestonSubject | SuggestionWork;
+
+export const AutosuggestText: React.FC<AutosuggestTextProps> = ({
+  responseData,
+  highlightedIndex,
+  getItemProps,
+  stringSuggestionAuthorText = "author",
+  stringSuggestionWorkText = "work",
+  stringSuggestionTopicText = "topic"
+}) => {
+  function itemToString(objectItem: Suggestion) {
+    switch (objectItem.__typename) {
+      case "Creator":
+        return objectItem.name;
+      case "Subject":
+        return objectItem.value;
+      default:
+        return objectItem.title;
+    }
+  }
+
+  function generateItemId(objectItem: Suggestion) {
+    return `${objectItem.__typename}-${itemToString(objectItem)}`;
+  }
+
+  return (
+    <>
+      {responseData.map((item, index) => {
+        return (
+          <>
+            {/* eslint-disable react/jsx-props-no-spreading */}
+            {/* The downshift combobox works this way by design */}
+            {/* the style prop below should be deleted once the new design system
+              npm library is added to the project as we are currently experiencing
+              gh action problems due to issues at github that are out of our hands 
+              -- autosuggest-text-container__item--highlight class does the 
+              highlighting once we have it from the npm package */}
+            <li
+              className={`autosuggest-text-container__item text-body-medium-regular ${
+                highlightedIndex === index
+                  ? "autosuggest-text-container__item--highlight"
+                  : ""
+              }`}
+              key={generateItemId(item)}
+              style={
+                highlightedIndex === index ? { backgroundColor: "#EEE9E5" } : {}
+              }
+              {...getItemProps({ item, index })}
+            >
+              {/* eslint-enable react/jsx-props-no-spreading */}
+              {item.__typename === "Creator"
+                ? `${item.name} (${stringSuggestionAuthorText})`
+                : null}
+              {item.__typename === "Subject"
+                ? `${item.value} (${stringSuggestionTopicText})`
+                : null}
+              {item.__typename === "Work"
+                ? `${item.title} (${stringSuggestionWorkText})`
+                : null}
```

Comment:
Please use the new `useText()` from #27 instead of prop drilling.

Reply:
We can do that later on. [#27](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/27) is not included in this branch and I would rather add that in a future feature/autosuggest-advanced branch than rebasing release/2 into this branch (since there for sure is gonna be a new branch for autosuggest)
Added it to [DDFSOEG-144](https://reload.atlassian.net/jira/software/c/projects/DDFSOEG/boards/457?modal=detail&selectedIssue=DDFSOEG-144) ticket